### ISSUE TITLE
conduit-parse-0.1.2.2 build fails with latest conduit

### ISSUE THREAD

**Issue by DanBurton** [k0ral/conduit-parse#2]
Seen on the Stackage build server; I was able to reproduce locally like so:

```bash
stack unpack conduit-parse-0.1.2.2 && cd conduit-parse-0.1.2.2
edit stack.yaml # add the following stack.yaml
stack build
```

```yaml
# stack.yaml
resolver: nightly-2018-03-10
extra-deps:
- conduit-1.3.0
- resourcet-1.2.0
- conduit-extra-1.3.0
- conduit-combinators-1.3.0
```

The build failure I get from this is:

```
[1 of 2] Compiling Data.Conduit.Parser.Internal ( Data/Conduit/Parser/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Conduit/Parser/Internal.o )

/Users/dan/scratch/conduit-parse-0.1.2.2/Data/Conduit/Parser/Internal.hs:43:1: error:
    • Could not deduce (MonadCatch (ConduitT i Void m))
        arising from a use of ‘exceptions-0.8.3:Control.Monad.Catch.catch’
      from the context: MonadCatch m
        bound by the instance declaration
        at Data/Conduit/Parser/Internal.hs:43:1-66
      or from: Exception e
        bound by the type signature for:
                   exceptions-0.8.3:Control.Monad.Catch.catch :: forall e a.
                                                                 Exception e =>
                                                                 ConduitParser i m a
                                                                 -> (e -> ConduitParser i m a)
                                                                 -> ConduitParser i m a
        at Data/Conduit/Parser/Internal.hs:43:1-66
      or from: Exception e1
        bound by a type expected by the context:
                   forall e1 a1.
                   Exception e1 =>
                   ExceptT
                     ConduitParserException (StateT ([Text], Buffer i) (Sink i m)) a1
                   -> (e1
                       -> ExceptT
                            ConduitParserException (StateT ([Text], Buffer i) (Sink i m)) a1)
                   -> ExceptT
                        ConduitParserException (StateT ([Text], Buffer i) (Sink i m)) a1
        at Data/Conduit/Parser/Internal.hs:43:1-66
    • In the third argument of ‘ghc-prim-0.5.1.1:GHC.Prim.coerce’, namely
        ‘exceptions-0.8.3:Control.Monad.Catch.catch’
      In the expression:
        ghc-prim-0.5.1.1:GHC.Prim.coerce
          @(forall (e :: TYPE ghc-prim-0.5.1.1:GHC.Types.LiftedRep)
                   (a :: TYPE ghc-prim-0.5.1.1:GHC.Types.LiftedRep).
            Exception e =>
            ExceptT ConduitParserException StateT (,) [] Text Buffer i Sink i m a
            -> (e
                -> ExceptT ConduitParserException StateT (,) [] Text Buffer i Sink i m a)
               -> ExceptT ConduitParserException StateT (,) [] Text Buffer i Sink i m a)
          @(forall (e :: TYPE ghc-prim-0.5.1.1:GHC.Types.LiftedRep)
                   (a :: TYPE ghc-prim-0.5.1.1:GHC.Types.LiftedRep).
            Exception e =>
            ConduitParser i m a
            -> (e -> ConduitParser i m a) -> ConduitParser i m a)
          exceptions-0.8.3:Control.Monad.Catch.catch
      In an equation for ‘exceptions-0.8.3:Control.Monad.Catch.catch’:
          exceptions-0.8.3:Control.Monad.Catch.catch
            = ghc-prim-0.5.1.1:GHC.Prim.coerce
                @(forall (e :: TYPE ghc-prim-0.5.1.1:GHC.Types.LiftedRep)
                         (a :: TYPE ghc-prim-0.5.1.1:GHC.Types.LiftedRep).
                  Exception e =>
                  ExceptT ConduitParserException StateT (,) [] Text Buffer i Sink i m a
                  -> (e
                      -> ExceptT ConduitParserException StateT (,) [] Text Buffer i Sink i m a)
                     -> ExceptT ConduitParserException StateT (,) [] Text Buffer i Sink i m a)
                @(forall (e :: TYPE ghc-prim-0.5.1.1:GHC.Types.LiftedRep)
                         (a :: TYPE ghc-prim-0.5.1.1:GHC.Types.LiftedRep).
                  Exception e =>
                  ConduitParser i m a
                  -> (e -> ConduitParser i m a) -> ConduitParser i m a)
                exceptions-0.8.3:Control.Monad.Catch.catch
      When typechecking the code for ‘exceptions-0.8.3:Control.Monad.Catch.catch’
        in a derived instance for ‘MonadCatch (ConduitParser i m)’:
        To see the code I am typechecking, use -ddump-deriv
   |
43 | deriving instance (MonadCatch m) => MonadCatch (ConduitParser i m)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by DanBurton** [comment]
I've removed conduit-parse and xml-conduit-parse from the Stackage nightly build plan, and unfortunately they won't appear in the initial LTS 11 release.

To get back into nightly, open a PR on the Stackage repo as usual. To get into LTS 11, open an issue on https://github.com/fpco/lts-haskell/issues.

**Comment by k0ral** [comment]
Fixed in 61bf9dbc76d7798ec5197757d5d27d246ee28d8d, released in [0.2.0.0](https://hackage.haskell.org/package/conduit-parse-0.2.0.0).
Thank you.
